### PR TITLE
e2e/cli: Migrate completions, whoami and projects tests to go

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR migrates completions.sh, whoami.sh and projects.sh  tests to go version.
These tests will be executed for various infra types.

### CODE REVIEW

File: test/extended/cli/project.go [hunk 1]
```diff
@@ -21,4 +27,111 @@ var _ = g.Describe("[sig-cli] oc project", func() {
...
+
+		out, err := oc.Run("auth", "can-i").Args("get", "pods", "--as="+testProjectUser, "-n", "missing-ns").Output()
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("no"))
+
+		_, err = oc.Run("auth", "can-i").Args("--list", "--as="+testProjectUser, "-n", "missing-ns").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("projects with invalid argument should fail")
+		out, err = oc.Run("projects").Args("test_arg").Output()
```
Comment (ingvagabund) [inline test/extended/cli/project.go:59]: Based on https://github.com/openshift/oc/blob/master/pkg/cli/projects/projects.go running `oc projects` queries projects. Meaning `[apigroup:project.openshift.io]` is used as well.

Comment (ardaguclu) [reply test/extended/cli/project.go:59]: Thanks. Done.

File: test/extended/util/annotate/generated/zz_generated.annotations.go [hunk 2]
```diff
@@ -1753,8 +1759,6 @@ var annotations = map[string]string{
 
 	"[Top Level] [sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/builds.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]": "test/cmd/builds.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]",
 
-	"[Top Level] [sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/completions.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]": "test/cmd/completions.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]",
```
Comment (ardaguclu) [inline test/extended/util/annotate/generated/zz_generated.annotations.go:1756]: @ingvagabund  completions only tests cli completions and as far as I can see, it does not depend on any apigroup?, I removed these apigroups in new completions tests.

File: test/extended/util/annotate/generated/zz_generated.annotations.go [hunk 3]
```diff
@@ -1807,8 +1809,6 @@ var annotations = map[string]string{
 
 	"[Top Level] [sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/volumes.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]": "test/cmd/volumes.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]",
 
-	"[Top Level] [sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/whoami.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]": "test/cmd/whoami.sh [apigroup:image.openshift.io][apigroup:config.openshift.io]",
```
Comment (ardaguclu) [inline test/extended/util/annotate/generated/zz_generated.annotations.go:1810]: also same in here. whoami only sends whoami request to server to get urls?

**Comment by ardaguclu** [comment]
@ingvagabund if you have any pending PR that might be conflicting with this PR, we can park this one. If not, could you PTAL?

File: test/extended/util/annotate/generated/zz_generated.annotations.go (hunk 2 above)
Comment (ingvagabund) [reply test/extended/util/annotate/generated/zz_generated.annotations.go:1756]: Even better :) I made some assumptions about the code when I was going through it which is not always bullet proof.

**Comment by ardaguclu** [comment]
/hold

to check the results of jobs are not required

**Pull request merged**